core/geometry/src/topology/Graph.ts

```diff
@@ -1203,14 +1235,16 @@ export class HalfEdge implements HalfEdgeUserData {
       return node0;
     if (Geometry.isSameCoordinate(dy, 0.0))
       return undefined;
+    // parametric equation of line is (1-t)y0 + ty1 which is equal to y at the intersection so
+    // (1-t)y0 + ty1 = y or t = (y-y0)/(y1-y0)
     return Geometry.conditionalDivideFraction(y - node0.y, dy);
   }
-
   /**
-   * * Compute fractional coordinates of the intersection of a horizontal line with an edge.
-   * * If the edge is horizontal return undefined (no test for horizontal at y!!!)
-   * * If the edge is not horizontal and y is between its end y's, return the fraction
-   * @param node0 Base node of edge
+   * Compute fractional coordinates of the intersection of a horizontal line with an edge.
+   * * If the edge is horizontal return undefined (no test for horizontal at y).
+   * * If the edge is not horizontal and fraction is inside 0..1, return the fraction.
```

Comment:
The non-horizontal edge case is already described in the first line.
```suggestion
   * Compute the fractional y-coordinate of the intersection of a horizontal line with an edge.
   * * If the edge is horizontal return `undefined`.
```